front/src/modules/simulationResult/components/SpeedSpaceChart/helpers.ts

```diff
@@ -0,0 +1,184 @@
+import { mmToKm, msToKmh, mToKm } from 'utils/physics';
+
+import { electricalProfilesDesignValues } from './consts';
+
+const formatSpeed = (simulationBase: ReportTrainV2) => {
+  const { positions, speeds } = simulationBase;
+  return speeds.map((value, index) => ({
+    position: {
+      start: mmToKm(positions[index]),
+    },
+    value: msToKmh(value),
+  }));
+};
+
+const formatEcoSpeeds = (simulationFinalOutput: ReportTrainV2) => {
+  const { positions, speeds } = simulationFinalOutput;
+  return speeds.map((value, index) => ({
+    position: {
+      start: mmToKm(positions[index]),
+    },
+    value: msToKmh(value),
+  }));
+};
+
+const formatStops = (operationalPoints: PathPropertiesFormatted['operationalPoints']) =>
+  operationalPoints.map(({ position, extensions }) => ({
+    position: {
+      start: mmToKm(position),
+    },
+    value:
+      extensions?.identifier && extensions?.sncf
+        ? `${extensions.identifier.name} ${extensions.sncf.ch !== ('' || '00') ? extensions.sncf.ch : ''}`
+        : '',
+  }));
+
+const formatElectrifications = (electrifications: PathPropertiesFormatted['electrifications']) =>
+  electrifications.map(({ electrificationUsage, start, stop }) => ({
+    position: {
+      start: mToKm(start),
+      end: mToKm(stop),
+    },
+    value: {
+      type: electrificationUsage.type,
+      voltage:
+        'voltage' in electrificationUsage
+          ? (electrificationUsage.voltage as ElectrificationValues['voltage'])
+          : undefined,
+      lowerPantograph:
+        'lower_pantograph' in electrificationUsage ? electrificationUsage.lower_pantograph : false,
+    },
+  }));
+
+const formatSlopes = (slopes: PositionData<'gradient'>[]) => {
+  let actualSlopes: LayerData<number>[] = [];
+  if (slopes)
+    actualSlopes = slopes.map((slope, index) => ({
+      position: {
+        start: mToKm(index === 0 ? 0 : slopes[index - 1].position),
+        end: mToKm(slope.position),
+      },
+      value: slope.gradient,
+    }));
+  return actualSlopes;
+};
+
+const isNotCompatible = (voltage: string, profile: string) =>
+  (voltage === '1500V' && profile?.startsWith('BB')) ||
+  (voltage === '25000V' && profile?.startsWith('AA'));
+
+const getProfileValue = (
+  electricalProfileType: 'profile' | 'no_profile',
+  profile: keyof typeof electricalProfilesDesignValues | null,
+  voltage?: '1500V' | '25000V'
+) => {
+  if (electricalProfileType === 'no_profile') {
+    return { electricalProfile: 'neutral' };
+  }
+  if (profile === null || (profile && voltage && isNotCompatible(voltage, profile))) {
+    return {
+      electricalProfile: 'incompatible',
+      color:
+        voltage === '1500V'
+          ? electricalProfilesDesignValues.F.color
+          : electricalProfilesDesignValues['20000V'].color,
+    };
+  }
+  return {
+    electricalProfile: profile,
+    ...electricalProfilesDesignValues[profile!],
+  };
+};
+
+const formatElectricalProfiles = (
+  simulation: SimulationResponseSuccess,
```

Comment:
Instead of formatting datas that has already been formatted, do you think you could reuse/adapt some of the utils.ts (which are all tested) in `application/operationalStudies` ?